address_book/internal/service/service.go

```diff
@@ -102,18 +109,23 @@ func (a *AddressBookService) UpdateContact(_ context.Context, in *pb.UpdateConta
 		}, nil
 	}
 
-	if _, ok := a.data.Load(in.Contact.Phone); !ok {
+	contact := models.Contact{
+		Phone:   in.Contact.Phone,
+		Name:    in.Contact.Name,
+		Address: in.Contact.Address,
+	}
+
+	err := a.db.Model(&contact).Updates(models.Contact{Phone: contact.Phone, Name: contact.Name, Address: contact.Address}).Error
```

Comment:
What we are going to update here, and which value will we put into a DB as a result? I don't see `where` condition here. You have to use some uniqueId to define the records going to be updated, or you have to declare old value and new value (less preferable way) in the update request

Reply:
```contact``` object above is created. ```Phone``` is used as a unique parameter, and ```Name``` and ```Address``` are changed.

Reply:
oh, it seems I missed the model definition. ok